src/coreclr/vm/eventing/eventpipe/ep-rt-coreclr.h

```diff
@@ -1134,8 +1135,13 @@ ep_rt_entrypoint_assembly_name_get_utf8 (void)
 		}
 	}
 
-	// fallback to the empty string if we can't get assembly info, e.g., if the runtime is
```

Comment:
Do we need both paths? Are there scenarios where the host won't set the assembly name?

Reply:
Our host will always set it - except for in non-app (component) scenarios, in which case there is no entry assembly so it is expected to be empty. However, if someone directly hosts coreclr, they don't need to set it - we discourage direct hosting and document it as obsolete, but there is still some functionality for which we don't have parity.